### PR TITLE
Implement register_investor extrinsic 

### LINKED ISSUE #470: Register Account Extrinsic

### PR DESCRIPTION
**Pull request by felixfaisal**
This PR implements the register_investor extrinsic that will reserve PDEX from an investor

### CODE REVIEW

File: pallets/polkadex-ido/src/lib.rs [hunk 1]
```diff
@@ -194,6 +214,29 @@ pub mod pallet {
     #[pallet::call]
     impl<T: Config> Pallet<T> {
 
+        /// Registers a new investor to allow participating in funding round.
+		///
+		/// # Parameters
+		///
+		/// * `origin`: Account to be registered as Investor
```
Comment (serhii-temchenko) [inline pallets/polkadex-ido/src/lib.rs:221]: Before you told that `origin` is property which present in all extrinsicses, but now it's here. Please be consistent. If it's not needed (IMO with it it's more common and better) - remove everywhere, if it's needed - add everywhere.

File: pallets/polkadex-ido/src/lib.rs [hunk 2]
```diff
@@ -194,6 +214,29 @@ pub mod pallet {
...
+		#[pallet::weight((10_000, DispatchClass::Normal))]
+		pub fn register_investor(origin: OriginFor<T>) -> DispatchResult {
```
Comment (serhii-temchenko) [inline pallets/polkadex-ido/src/lib.rs:223]: In IDO documentation in notion name is different, please update one side...

File: pallets/polkadex-ido/src/lib.rs [hunk 3]
```diff
@@ -194,6 +214,29 @@ pub mod pallet {
...
+			let current_block_no = <frame_system::Pallet<T>>::block_number();
+			let who: T::AccountId = ensure_signed(origin)?;
+			ensure!(!<InfoInvestor<T>>::contains_key(&who), Error::<T>::InvestorAlreadyRegistered);
+			let amount: BalanceOf<T> = T::IDOPDXAmount::get();
+			ensure!(
+				T::Currency::reserve(&who, amount).is_ok(),
+				Error::<T>::FailedToMoveBalanceToReserve
+			);
+            // TODO: We are reserving for default 100 blocks, This must be discussed
```
Comment (serhii-temchenko) [inline pallets/polkadex-ido/src/lib.rs:232]: Where/when this must be discussed? Please discuss those questions before moving to `ReadyForReview` status.

File: pallets/polkadex-ido/src/lib.rs [hunk 4]
```diff
@@ -418,6 +463,11 @@ pub mod pallet {
         }
     }
 
+    /// Stores investor Info
+	#[pallet::storage]
+	#[pallet::getter(fn get_investorinfo)]
+	pub(super) type InfoInvestor<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, InvestorLockData<T>, OptionQuery>;
```
Comment (serhii-temchenko) [inline pallets/polkadex-ido/src/lib.rs:469]: Should be saved invested amount?

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/polkadex-ido/src/lib.rs (hunk 3 above)
Comment (Gauthamastro) [inline pallets/polkadex-ido/src/lib.rs:232]: There is no time limit for reserving for IDO; it will be reserved until the Investor deregisters.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

**Pull request closed without merging**